### ISSUE TITLE
Questions about ddppo

### ISSUE THREAD

**Issue by RichterLee** [facebookresearch/sound-spaces#83]
![image](https://user-images.githubusercontent.com/47584381/181609712-a49f4611-b9f5-4ae2-afd1-ec3c78d46f53.png)

Hi Changan,

Just a quick question about DDPPO algo in your code. Could you please tell me how distribe works here? If I want to add additional loss , how shall I change the code? Should I change to something like
                stats = torch.tensor(
                    [value_loss, action_loss, additional_loss, dist_entropy, count_steps_delta],
                    device=self.device,
                )
then call distrib.all_reduce(stats)?

Does the order of these tensor matters here? I just saw you have a state_ordering for the running episode.

Thanks you so much!!

**Comment by ChanganVR** [comment]
Hi @RichterLee, yes the process should be adding another tensor the end of the `stats` list and then calling the reduce function. The order here matters. It shouldb be pretty easy to figure out if you play with the setting a bit.

**Comment by RichterLee** [comment]
Many thanks to your help! Sorry that I'm still a little confused about the order of these tensors. I also read your savi code. I found that you add belief loss and accuracy after the dist_entropy and before the count_steps_delta. Is there a reason for such order?
![image](https://user-images.githubusercontent.com/47584381/182042216-26ab875a-f3ef-4606-9b50-e9f584c1f206.png)
